File: Coding/Dirichlet's theorem/dirichlet.cpp
// C++ implementation of
// the above approach
#include <bits/stdc++.h>
using namespace std;

// Utility function to check
// whether two numbers is
// co-prime or not
int coprime(int a, int b)
{
	if (__gcd(a, b) == 1)
		return true;
	else
		return false;
}

// Utility function to check
// whether a number is prime
// or not
bool isPrime(int n)
{
	// Corner case
	if (n <= 1)
		return false;

	if (n == 2 or n == 3)
		return true;

	// Check from 2 to sqrt(n)
	for (int i = 2; i * i <= n; i++)
		if (n % i == 0)
			return false;

	return true;
}

// finding the Prime numbers
void findNumbers(int a, int b, int n)
{

	bool possible = true;

	// Checking whether given
	// numbers are co-prime
	// or not
	if (!coprime(a, b))
		possible = false;

	int c1 = 1;
	int c2 = 1;

	int num1, num2;

	// To store the N primes
	set<int> st;
	// If 'possible' is true
	if (possible) {

		// Printing n numbers
		// of prime
		while ((int)st.size() != n) {

			// checking the form of a+nb
			num1 = a + (c1 * b);
			if (isPrime(num1)) {
				st.insert(num1);
			}
			c1++;

			// Checking the form of b+na
			num2 = b + (c2 * a);
			if (isPrime(num2)) {
				st.insert(num2);
			}
			c2++;
		}

		for (int i : st)
			cout << i << " ";
	}

	// If 'possible' is false
	// return -1
	else
		cout << "-1";
}

// Driver Code
int main()
{

	int a = 2;
	int b = 1;
	int n = 10;

	findNumbers(a, b, n);
	return 0;
}

File: Coding/Dirichlet's theorem/dirichlet.py
# Python3 implementation of the above approach
from math import gcd, sqrt
from sympy import Q, ask

# Utility function to check
# whether two numbers is
# co-prime or not
def coprime(a, b) :
	
	if (gcd(a, b) == 1) :
		return True;
		
	else :
		return False;

# finding the Prime numbers
def findNumbers(a, b, n) :

	possible = True;

	# Checking whether given
	# numbers are co-prime
	# or not
	if (not coprime(a, b)) :
		possible = False;

	c1 = 1;
	c2 = 1;

	num1 = 0;
	num2 = 0;

	# To store the N primes
	st = set();
	
	# If 'possible' is true
	if (possible) :

		# Printing n numbers
		# of prime
		while (len(st) != n) :

			# checking the form of a+nb
			num1 = a + (c1 * b);
			
			if ask(Q.prime(num1)):
				
				st.add(num1);
				
			c1 += 1;

			# Checking the form of b+na
			num2 = b + (c2 * a);
			
			if ask(Q.prime(num2)):
				st.add(num2);
	
			c2 += 1;

		for i in st :
			print(i, end = ", ");

	# If 'possible' is false
	# return -1
	else :
		print("-1");

#  	A065091 - OEIS sequence
# 2n + 1
# Driver Code
if __name__ == "__main__" :

	a = 2;
	b = 1;
	n = 10;

	findNumbers(a, b, n);

# This code is contributed by AnkitRai01
